tap_mssql/__init__.py

```diff
@@ -632,9 +649,8 @@ def sync_non_cdc_streams(mssql_conn, non_cdc_catalog, config, state):
                 f"No replication key for {catalog_entry.table}, using full table replication"
             )
             replication_method = "FULL_TABLE"
-        if replication_method == "INCREMENTAL" and not primary_keys:
-            LOGGER.info(f"No primary key for {catalog_entry.table}, using full table replication")
-            replication_method = "FULL_TABLE"
+        # Check for INCREMENTAL load without primary keys removed

```

Comment:
Resolving issue that you can do an incremental with just a replication_key specified. For this reason, this existing test was removed.